shell/platform/android/io/flutter/embedding/android/FlutterImageView.java

```diff
@@ -62,29 +82,56 @@ public FlutterRenderer getAttachedRenderer() {
    */
   @Override
   public void attachToRenderer(@NonNull FlutterRenderer flutterRenderer) {
-    if (this.flutterRenderer != null) {
-      this.flutterRenderer.stopRenderingToSurface();
-    }
-
     this.flutterRenderer = flutterRenderer;
-    flutterRenderer.startRenderingToSurface(imageReader.getSurface());
+    switch (kind) {
+      case background:
+        flutterRenderer.swapSurface(imageReader.getSurface());
+        break;
+      case overlay:
+        // Don't do anything as this is done by the handler of
+        // `FlutterJNI#createOverlaySurface()` in the native side.
+        break;
+    }
   }
 
   /**
    * Invoked by the owner of this {@code FlutterImageView} when it no longer wants to render a
    * Flutter UI to this {@code FlutterImageView}.
    */
   public void detachFromRenderer() {
-    if (flutterRenderer != null) {
-      flutterRenderer.stopRenderingToSurface();
-      flutterRenderer = null;
+    switch (kind) {
+      case background:
+        // TODO: Swap the surface back to the original one.
+        // https://github.com/flutter/flutter/issues/58291
+        break;
+      case overlay:
+        // TODO: Handle this in the native side.
+        // https://github.com/flutter/flutter/issues/59904
+        break;
     }
   }
 
+  public void pause() {
+    // Not supported.
+  }
+
   /** Acquires the next image to be drawn to the {@link android.graphics.Canvas}. */
   @TargetApi(19)
   public void acquireLatestImage() {
-    nextImage = imageReader.acquireLatestImage();
+    // There's no guarantee that the image will be closed before the next call to
+    // `acquireLatestImage()`. For example, the device may not produce new frames if
+    // it's in sleep mode, so the calls to `invalidate()` will be queued up
+    // until the device produces a new frame.
+    //
+    // While the engine will also stop producing frames, there is a race condition.
+    //
+    // To avoid exceptions, check if a new image can be acquired.
+    if (pendingImages < imageReader.getMaxImages()) {
```

Comment:
To repro this bug: while running the app, turn off the screen and the app crashes

Reply:
Does the error still happen if you increase maxImages as done in https://github.com/flutter/engine/pull/19319?

See Android's recommendation in https://developer.android.com/reference/android/media/ImageReader#acquireLatestImage()

Reply:
yeah. `onDraw()` it throttled or not called at all, which leaves the images open.

Reply:
yeah. onDraw() is throttled or not called at all, which leaves the images open.

Reply:
Hi, I faced this crash issue recently. I'd like to ask do we have any solution for this?

Reply:
Please refer to this issue link https://github.com/flutter/flutter/issues/63897. Thanks.